### ISSUE TITLE
Learn words representation with CBOW plus position-weights

### ISSUE THREAD

**Issue by loretoparisi** [facebookresearch/fastText#445]
As for the recent paper "Learning Word Vectors for 157 Languages", the model CBOW is used with position dependent weights. Using that the new pre-trained model were produced.
Is it possible to train a unsupervised model with CBOW in this version of fastText using the same approach with position weights?

**Comment by rilut** [comment]
Also, which command in fasttext corresponds to the positional-weighting?

**Comment by matanster** [comment]
I don't think this is implemented in the publicly available codebase.

**Comment by loretoparisi** [comment]
@matanster you are right, it is not, at least not yet, while the most two recent papers show models that has been trained FastText with position dependent weights.

**Comment by omriFdna** [comment]
Do you know if there are any updates of positional-weighting in fasttext base code?

**Comment by loretoparisi** [comment]
@omriFdna so far I'm not aware of a version that implements the positional weights bow, but I will have a look maybe someone did...

**Comment by omriFdna** [comment]
@loretoparisi any luck?

**Comment by loretoparisi** [comment]
@omriFdna not yet so far :

**Comment by loretoparisi** [comment]
A dataset with position weights trained with CBOW has been recently released for Wikipedia and Common Crawl - https://fasttext.cc/docs/en/crawl-vectors.html

The model parameters were:

|DIM|NGRAM|WS|NEG
|---|---|---|---|
|300|5|5|10| 


It would be great to have this as training option as well.

**Comment by omriFdna** [comment]
I agree, they also report in their papers that position weights improves the performance, I wish it was part of the training options.

**Comment by adam2326** [comment]
+1

**Comment by Witiko** [comment]
I am attempting to add this feature to Gensim, see https://github.com/RaRe-Technologies/gensim/pull/2905.

**Comment by Witiko** [comment]
The work from https://github.com/RaRe-Technologies/gensim/pull/2905 has been accepted as [a journal paper][1] that is to appear in [J.UCS 28:2][4]. To conduct experiments for the paper, I have produced [the high-level PInE library][2] that uses [a fork of Gensim 3.8.3][3] for model training. Perhaps the PInE library can serve as an inspiration for an implementation to facebook's fastText and to current Gensim. 💡 

 [1]: https://arxiv.org/abs/2104.09691
 [2]: https://github.com/MIR-MU/pine
 [3]: https://github.com/witiko/gensim/tree/pine
 [4]: https://www.jucs.org/jucs_journal.html
